--- heartbeat-1.2.3.debian/configure.in	2004-09-18 15:24:22.000000000 +0900
+++ heartbeat-1.2.2.debian/configure.in	2004-08-29 17:13:23.000000000 +0900
@@ -1173,6 +1134,40 @@
 AC_DEFINE_UNQUOTED(UCDSNMPDEVEL, $UCDSNMPDEVEL, 
 	Name of UCD SNMP development package)
 
+#
+#	Path to SSH
+#       This is guessed if ssh is installed.
+#       However, there is a long standing bug with installing ssh
+#       on arm in debian and this is preventing heartbeat from
+#       moving into sarge. By adding this we can remove the build
+#       dependancy on ssh and get an updated heartbeat into sarge... 
+#       I hope. Horms 26th July 2004
+#
+AC_ARG_WITH(
+  ssh-path,
+    [  --with-ssh-path=path    Path to SSH binary (e.g. /usr/bin/ssh)
+                          [default=autodetect]  ],
+    [ SSH=$withval ],
+  )
+AC_SUBST(SSH)
+
+#
+#	Path to SCP
+#       This is guessed if ssh is installed.
+#       However, there is a long standing bug with installing ssh
+#       on arm in debian and this is preventing heartbeat from
+#       moving into sarge. By adding this we can remove the build
+#       dependancy on ssh and get an updated heartbeat into sarge... 
+#       I hope. Horms 26th July 2004
+#
+AC_ARG_WITH(
+  scp-path,
+    [  --with-scp-path=path    Path to SCP binary (e.g. /usr/bin/scp)
+                          [default=autodetect]  ],
+    [ SCP=$withval ],
+  )
+AC_SUBST(SCP)
+
 dnl ************************************************************************
 dnl checks for library functions to replace them
 dnl
