Index: Makefile.am
===================================================================
RCS file: /home/cvs/linux-ha/linux-ha/Makefile.am,v
retrieving revision 1.61
diff -u -r1.61 Makefile.am
--- a/Makefile.am	11 Jul 2005 16:36:41 -0000	1.61
+++ b/Makefile.am	14 Jul 2005 12:51:09 -0000
@@ -53,11 +53,15 @@
 
 ## proc-ha is left out from SUBDIRS (completely obsolete)
 
+if SNMP_SUBAGENT_BUILD
+SNMP_SUBAGENT_DIR = snmp_subagent
+endif
+
 SUBDIRS	=		$(LIBLTDL_DIR) debian pkg port replace include config	\
 			lib						\
 			ldirectord heartbeat membership telecom		\
 			linux-ha contrib resources			\
-			crm lrm snmp_subagent				\
+			crm lrm $(SNMP_SUBAGENT_DIR)			\
 			tools logd fencing doc cts
 
 HANDY_DOCS = doc/ChangeLog doc/GettingStarted.html doc/DirectoryMap.txt
Index: lib/bindings/perl/Makefile.am
===================================================================
RCS file: /home/cvs/linux-ha/linux-ha/lib/bindings/perl/Makefile.am,v
retrieving revision 1.11
diff -u -r1.11 Makefile.am
--- a/lib/bindings/perl/Makefile.am	2 Jun 2005 03:14:18 -0000	1.11
+++ a/lib/bindings/perl/Makefile.am	7 Jul 2005 06:00:12 -0000
@@ -19,17 +19,16 @@
 #
 
 
-EXTRA_DIST =	.cvsignore Makefile.PL.in MANIFEST test.pl 	\
-		cl_raw/.cvsignore				\
-		cl_raw/Makefile.PL.in cl_raw/Makefile.PL 	\
-		cl_raw/cl_log_wrappers.c			\
-		cl_raw/cl_log_wrappers.h  cl_raw/cl_raw.i 	\
-		cl_raw/ipc_wrappers.c cl_raw/ipc_wrappers.h	\
-		lib/heartbeat/clplumbing/ipc.pm			\
+EXTRA_DIST =	.cvsignore Makefile.PL.in MANIFEST test.pl \
+		lib/heartbeat/clplumbing/ipc.pm \
 		lib/heartbeat/clplumbing/log.pm
 
 MAINTAINERCLEANFILES    = Makefile.in
 
+if SWIG_BUILD
+SWIG_DIR = cl_raw
+endif
+
 ## Subdirectories...
 SUBDIRS			=  	\
-	cl_raw
+	$(SWIG_DIR)
Index: lib/bindings/perl/cl_raw/Makefile.am
===================================================================
RCS file: /home/cvs/linux-ha/linux-ha/lib/bindings/perl/cl_raw/Makefile.am,v
retrieving revision 1.1
diff -u -r1.1 Makefile.am
--- a/lib/bindings/perl/cl_raw/Makefile.am	1 Jun 2005 14:51:29 -0000	1.1
+++ a/lib/bindings/perl/cl_raw/Makefile.am	7 Jul 2005 06:00:12 -0000
@@ -18,19 +18,16 @@
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 #
 
-if SWIG_BUILD
-
 noinst_DATA = .made_mod
 
 MAINTAINERCLEANFILES    = Makefile.in
 
 EXTRA_DIST      = \
+	.cvsignore \
 	Makefile.PL.in Makefile.PL \
 	cl_log_wrappers.c cl_log_wrappers.h \
 	cl_raw.i \
-	ipc_wrappers.c ipc_wrappers.h \
-	lib/heartbeat/clplumbing/ipc.pm \
-	lib/heartbeat/clplumbing/log.pm
+	ipc_wrappers.c ipc_wrappers.h
 
 INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include \
 	-I$(top_builddir)/linux-ha -I$(top_srcdir)/linux-ha  \
@@ -63,5 +60,3 @@
 clean-local:
 	if test -f Makefile.perl; then $(MAKE) -fMakefile.perl realclean; fi
 	rm -f Makefile.perl Makefile.perl.old
-
-endif
Index: snmp_subagent/Makefile.am
===================================================================
RCS file: /home/cvs/linux-ha/linux-ha/snmp_subagent/Makefile.am,v
retrieving revision 1.20
diff -u -r1.20 Makefile.am
--- a/snmp_subagent/Makefile.am	24 May 2005 22:29:34 -0000	1.20
+++ a/snmp_subagent/Makefile.am	7 Jul 2005 06:00:12 -0000
@@ -50,7 +50,6 @@
 mibs_DATA		= LINUX-HA-MIB.mib
 DIST_LIST		= SNMPAgentSanityCheck.in $(mibs_DATA)
 
-if SNMP_SUBAGENT_BUILD
 EXTRA_DIST		= $(DIST_LIST)
 
 
@@ -81,7 +80,3 @@
 # test cases
 testdir			= $(libdir)/heartbeat
 test_SCRIPTS		= $(test_scrs)
-
-else
-EXTRA_DIST		= $(DIST_LIST)	$(hbagent_srces) $(test_scrs)
-endif
