Index: Makefile.am
===================================================================
RCS file: /home/cvs/linux-ha/linux-ha/Makefile.am,v
retrieving revision 1.61
diff -u -r1.61 Makefile.am
--- a/Makefile.am	11 Jul 2005 16:36:41 -0000	1.61
+++ b/Makefile.am	14 Jul 2005 12:51:09 -0000
@@ -53,11 +53,15 @@
 
 ## proc-ha is left out from SUBDIRS (completely obsolete)
 
+if SNMP_SUBAGENT_BUILD
+SNMP_SUBAGENT_DIR = snmp_subagent
+endif
+
 SUBDIRS	=		$(LIBLTDL_DIR) debian pkg port replace include config	\
 			lib						\
 			ldirectord heartbeat membership telecom		\
 			linux-ha contrib resources			\
-			crm lrm snmp_subagent				\
+			crm lrm $(SNMP_SUBAGENT_DIR)			\
 			tools logd fencing doc cts
 
 HANDY_DOCS = doc/ChangeLog doc/GettingStarted.html doc/DirectoryMap.txt
Index: snmp_subagent/Makefile.am
===================================================================
RCS file: /home/cvs/linux-ha/linux-ha/snmp_subagent/Makefile.am,v
retrieving revision 1.20
diff -u -r1.20 Makefile.am
--- a/snmp_subagent/Makefile.am	24 May 2005 22:29:34 -0000	1.20
+++ a/snmp_subagent/Makefile.am	7 Jul 2005 06:00:12 -0000
@@ -50,7 +50,6 @@
 mibs_DATA		= LINUX-HA-MIB.mib
 DIST_LIST		= SNMPAgentSanityCheck.in $(mibs_DATA)
 
-if SNMP_SUBAGENT_BUILD
 EXTRA_DIST		= $(DIST_LIST)
 
 
@@ -81,7 +80,3 @@
 # test cases
 testdir			= $(libdir)/heartbeat
 test_SCRIPTS		= $(test_scrs)
-
-else
-EXTRA_DIST		= $(DIST_LIST)	$(hbagent_srces) $(test_scrs)
-endif
