--- a/heartbeat/lib/ResourceManager.in	2004-11-10 12:15:50.000000000 +0900
+++ b/heartbeat/lib/ResourceManager.in	2004-11-10 12:16:04.000000000 +0900
@@ -183,7 +183,7 @@
   case `$spath $arg status` in
     # uncomment next line if you have misbehaving scripts,
     # and you don't want to fix each single one of them...
-    #  *not\ [Rr]unning*)	return 3;;
+    *not\ [Rr]unning*)	return 3;;
     *[Rr]unning*|*OK*)	return 0;;
     *)			return 3;;
   esac
