--- a/configure.in	2005-09-15 15:36:30.000000000 +0900
+++ b/configure.in	2005-09-15 16:34:07.000000000 +0900
@@ -316,11 +316,6 @@
 dnl ************************************************************************
 AC_PATH_PROGS(MKTEMP, mktemp)
 
-if test -n "$WHOAMI"; then
-  IAM=`$WHOAMI`
-fi
-AC_MSG_CHECKING(if chown works for non-root)
-
 dnl Prefer "mktemp" command.  But some OSes lack it; they can "touch".
 if test -n "$MKTEMP"; then
   F=`$MKTEMP "./.chown-testXXXXX"`
@@ -329,17 +324,33 @@
   touch $F
 fi
 
-if
+AC_MSG_CHECKING(if chown works for non-root)
+if test -n "$WHOAMI"  ; then
+  IAM=`$WHOAMI`
+fi
+# In the cae of Debian, chown to root does not work as non-root
+# And to futher complicate things, the package is often
+# built with fakeroot, which looks like root, but issn't,
+# causing the test to stall waiting for nobody's password
+# Resolve this by checking if fakeroot is in operation,
+# and if it is assuming the system is Debian and failing
+# the test
+nonroot_chown=no
+if 
+  test "X$FAKEROOTKEY" = "X" &&
   case "$IAM" in
     root)	chown nobody $F;
-       		su nobody -c "chown root $F";;
+     		  su nobody -c "chown root $F";;
     *)	chown root $F
   esac >/dev/null 2>&1
 then
   nonroot_chown=yes
+fi
+if 
+  test "$nonroot_chown" = yes
+then
   AC_MSG_RESULT(yes)
 else
-  nonroot_chown=no
   AC_MSG_RESULT(no)
 fi
 rm -f $F
