--- a/telecom/recoverymgrd/Makefile.am	2004-11-10 16:19:51.000000000 +0900
+++ b/telecom/recoverymgrd/Makefile.am	2004-11-10 16:21:22.000000000 +0900
@@ -72,3 +72,20 @@
 recoverymgrd_LDFLAGS	= @LIBADD_DL@ @LIBLTDL@ -export-dynamic		\
 			-dlpreopen-force 				\
 			-dlopen-self
+
+#
+#       This seems to be needed to compile on Debian Sarge (10 Nov 2004)
+#       Bison 1.875a (1:1.875a-1.1)
+#       Flex 2.5.31  (2.5.31-27)
+#
+lex.yy.$(OBJEXT): lex.yy.c
+	old_pwd=`pwd` ; cd $(srcdir) &&				\
+	$(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS)	\
+	-c lex.yy.c						\
+	-o $$old_pwd/$@  $(NON_FATAL_CFLAGS) -Wall
+
+conf_lex.$(OBJEXT): conf_lex.c
+	old_pwd=`pwd` ; cd $(srcdir) &&				\
+	$(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS)	\
+	-c conf_lex.c						\
+	-o $$old_pwd/$@  $(NON_FATAL_CFLAGS) -Wall
