--- heartbeat-1.2.3.orig/configure.in
+++ heartbeat-1.2.3/configure.in
@@ -1848,10 +1882,10 @@
 
 case x"$SNMP_SUBAGENT_DIR" in
   x)
-    AC_MSG_RESULT([  Build snmp subagent  = no])
+    AC_MSG_RESULT([  Build snmp subagent  = "no"])
     ;;
   xsnmp_subagent)
-    AC_MSG_RESULT([  Build snmp subagent  = yes])
+    AC_MSG_RESULT([  Build snmp subagent  = "yes"])
     AC_MSG_RESULT([  ucd_snmp_devel name  = "${UCDSNMPDEVEL}"])
     AC_MSG_RESULT([  SNMP libraries       = "${SNMP_LIBS}"])
     AC_MSG_RESULT([  SNMP MIB directory   = "${MIBS_DIR}"])
