Index: ./resources/heartbeat/LVM.in
===================================================================
RCS file: /home/cvs/linux-ha/linux-ha/resources/heartbeat/LVM.in,v
retrieving revision 1.1
diff -u -r1.1 LVM.in
--- ./resources/heartbeat/LVM.in	27 Aug 2004 09:34:14 -0000	1.1
+++ ./resources/heartbeat/LVM.in	26 Jul 2005 06:23:17 -0000
@@ -102,8 +102,15 @@
 #
 LVM_status() {
 
-  vgdisplay $1 | grep -i 'Status.*available' >/dev/null
-  
+  if 
+    [ "$LVM_MAJOR" -eq "1" ]
+  then
+  	vgdisplay $1 | grep -i 'Status.*available' >/dev/null
+	return $?
+  else
+  	vgdisplay -v $1 | grep -i 'Status[ \t]*available' &>/dev/null
+	return $?
+  fi
 }
 
 #
@@ -111,15 +118,27 @@
 #
 LVM_report_status() {
 
-  VGOUT=`vgdisplay $1 2>&1`
+  if 
+    [ "$LVM_MAJOR" -eq "1" ]
+  then
+  	VGOUT=`vgdisplay $1 2>&1`
+  	echo "$VGOUT" | grep -i 'Status.*available' >/dev/null
+	rc=$?
+  else
+  	VGOUT=`vgdisplay -v $1 2>&1`
+	echo "$VGOUT" | grep -i 'Status[ \t]*available' >/dev/null
+	rc=$?
+  fi
+
   if
-    echo "$VGOUT" | grep -i 'Status.*available' >/dev/null
+    [ $rc -eq 0 ]
   then
     : Volume $1 is available
   else
     echo "LVM Volume $1 is not available (stopped)"
     return 1
   fi
+
   if
     echo "$VGOUT" | grep -i 'Access.*read/write' >/dev/null
   then
@@ -155,7 +174,15 @@
 LVM_start() {
 
   ha_log "Activating volume group $1"
-  run vgscan $1
+
+  if 
+    [ "$LVM_MAJOR" -eq "1" ]
+  then
+  	run vgscan $1
+  else
+  	run vgscan
+  fi
+
   run vgchange -a y $1 || return 1
 
   if
@@ -198,6 +225,18 @@
   exit $?
 fi
 
+LVM_VERSION=`vgchange --version 2>&1 | \
+	awk --source '/Logical Volume Manager/ {print $5"\n"; exit; } '\
+	    --source '/LVM version:/ {printf $3"\n"; exit;}'`
+rc=$?
+if 
+  [ $rc -ne 0 ]
+then
+  ha_log "ERROR: LVM: $1 could not determine LVM version"
+  return  $rc
+fi
+LVM_MAJOR="${LVM_VERSION%%.*}"
+
 VOLUME=$1
 
 # What kind of method was invoked?
