Index: configure.in
===================================================================
RCS file: /home/cvs/linux-ha/linux-ha/configure.in,v
retrieving revision 1.348
diff -u -r1.348 configure.in
--- a/configure.in	8 Nov 2004 20:48:36 -0000	1.348
+++ b/configure.in	10 Nov 2004 09:12:02 -0000
@@ -245,23 +245,11 @@
 AC_PATH_PROGS(DRBDSETUP, drbdsetup)
 AC_PATH_PROGS(PKGCONFIG, pkg-config)
 
-WHOAMI=whoami
 AC_MSG_CHECKING(if chown works for non-root)
-F=`mktemp "./.chown-testXXXXX"`
-if
-  case `$WHOAMI` in
-    root)	chown nobody $F;
-       		su nobody -c "chown root $F";;
-    *)	chown root $F
-  esac >/dev/null 2>&1
-then
-  nonroot_chown=yes
-  AC_MSG_RESULT(yes)
-else
-  nonroot_chown=no
-  AC_MSG_RESULT(no)
-fi
-rm -f $F
+# There is a test for this upstream, but it does not work
+# at all well with fakeroot
+nonroot_chown=no
+AC_MSG_RESULT(no)
 AM_CONDITIONAL(NONROOT_CHOWN, test "$nonroot_chown" = yes )
 
 
